assets/persistence/example_data.gd

```diff
@@ -0,0 +1,22 @@
+extends PersistentDataSection
+
+const PD_SECTION_EXAMPLE = "example"
+const PD_SECTION_EXAMPLE_TIME = "time"
+const PD_SECTION_EXAMPLE_VALUE = "value"
+
+var time: int
+var value: int
+
+func get_tag() -> String:
+	return PD_SECTION_EXAMPLE
+
+func serialise() -> Dictionary:
```

Comment:
Is there any typing on this Dictionaries values? Can they be anything, or only strings/ castable to a string?

Reply:
Literally anything, unfortunately. Only `Array` currently supports typing with the `Array[int]`. There is, however, a current proposal for doing a similar thing with Dictionaries (`Dictionary[int, String]`) planned for Godot 4.x here: https://github.com/godotengine/godot-proposals/issues/56. I'm hoping we can add those in when available, but I should note that this system is 1) a candidate for moving to a C++ plugin eventually anyway, and 2) will spit out several errors if you try to provide non-JSON serialisable keys and/or values at runtime so I think we're safe, it's just not nice.